File: a024(gcd)/C++/gcd.cpp
#include <iostream>

int gcd(int a, int b) {
  if(b) {
    return gcd(b, a % b);
  }
  return a;
}


int main(void) {
  int a, b;
  while (std::cin >> a >> b) {
    std::cout << (gcd(a, b)) << '\n';
  }

  return 0;
}

File: a024(gcd)/Python/gcd.py

def gcd(a: int, b: int) -> int:
    if b:
        return gcd(b, a % b)
    return a;

while True:
    try:
        a, b = map(int, input().split())
        print(gcd(a, b))
    except:
        break
